### PR TITLE
[telco5g] Ci use prepared cluster

### PR DESCRIPTION
**Pull request by gpuckett-redhat**
cuts down on wasted time during testing

### CODE REVIEW

File: ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh [hunk 1]
```diff
@@ -65,6 +65,13 @@ cat << EOF > $SHARED_DIR/get-cluster-name.yml
     register: cluster
     environment:
       JOB_NAME: ${JOB_NAME:-'unknown'}
+    when: cnftests == "false"
```
Comment (sshnaidm) [inline ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh:68]:
```suggestion
    when: not cnftests|bool
```

File: ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh [hunk 2]
```diff
@@ -65,6 +65,13 @@ cat << EOF > $SHARED_DIR/get-cluster-name.yml
...
+  - name: Get prepared cluster parameters
+    command: python3 ~/telco5g-lab-deployment/scripts/upstream_cluster_all.py --get-cluster --cluster-name cnfdc2 --force
+    register: cluster
+    environment:
+      JOB_NAME: ${JOB_NAME:-'unknown'}
+    when: cnftests == "true"
```
Comment (sshnaidm) [inline ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh:74]:
```suggestion
    when: cnftests|bool
```

File: ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh [hunk 3]
```diff
@@ -232,6 +245,8 @@ cat << EOF > ~/fetch-information.yml
     shell: kcli ssh root@${CLUSTER_NAME}-installer 'oc get node'
 EOF
 
+if [[ "$T5CI_JOB_TYPE" != "cnftests" ]]; then
 ANSIBLE_STDOUT_CALLBACK=debug ansible-playbook -i $SHARED_DIR/inventory ~/ocp-install.yml -vv
```
Comment (sshnaidm) [inline ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh:249]: please add indentation

File: ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh [hunk 4]
```diff
@@ -74,7 +81,13 @@ EOF
 ping ${BASTION_IP} -c 10 || true
 echo "exit" | curl telnet://${BASTION_IP}:22 && echo "SSH port is opened"|| echo "status = $?"
 
-ansible-playbook -i $SHARED_DIR/bastion_inventory $SHARED_DIR/get-cluster-name.yml -vvvv
+CNF_ARG=""
```
Comment (sshnaidm) [inline ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh:84]: That's good for the first iteration. Please also check an option to manipulate actual parameters for upstream_cluster_all.py - for example:
```bash
ADDITIONAL_ARG=""
if [[ "$T5CI_JOB_TYPE" == "cnftests" ]]; then
  ADDITIONAL_ARG="--cluster-name cnfdc2 --force"
fi
```
And then just using the same task:

```yaml
- name: Discover cluster to run job
  command: python3 ~/telco5g-lab-deployment/scripts/upstream_cluster_all.py --get-cluster -e $CL_SEARCH $ADDITIONAL_ARG
  register: cluster
  environment:
    JOB_NAME: ${JOB_NAME:-'unknown'}
```
Then we leave ansible execution as is and changing just the params inside.

**Review by sshnaidm — CHANGES_REQUESTED** [review]
Please rebase on latest branch

**Comment by gpuckett-redhat** [comment]
bad rebase, closing this one

**Pull request closed without merging**